### ISSUE TITLE
Zooming in to a map on IE7 compatibility mode.

### ISSUE THREAD

**Issue by PaulDalek** [highcharts/highcharts#5412]
#### Behaviour

Zooming in to the map in this demo: http://www.highcharts.com/maps/demo/color-axis in IE7 compatibility mode causes a crash. This is the place in code where crash occurs: https://github.com/highcharts/highmaps-release/blob/v4.2.5/highmaps.src.js#L3121. On IE10, IE9 and IE8 zoom feature works fine.
#### Affected browser(s)

IE7

**Comment by TorsteinHonsi** [comment]
I'm not able to reproduce it right now, but apparently the issue is related to this regex:

```js
		if (/^[\-0-9\.]+$/.test(ret)) { // is numerical
			ret = parseFloat(ret);
		}
```

Most likely, this snippet can be replaced by a double-equal comparison, which may also be faster:

```js
		var parsed = parseFloat(ret);
		if (parsed == ret) { // eslint-disable-line eqeqeq
			ret = parsed;
		}
```

**Comment by PaulDalek** [comment]
Hi,

I created a wrapper, replacing regex test with double-equal comparison as listed above, and unfortunately I got the same error, but I managed to find out something helpful. It turns out that in IE7 (compatibility mode in IE11), zooming is followed by return of an empty object, which can’t be parsed by parseFloat() method. That’s why crash did occur before. I added a simple if, which checks if variable ret isn’t empty object. If so, the rest code execute normal. If it is, 0 is returned. As I mentioned before, I tested this only in IE7 compatibility mode in IE11, so I don’t know how it will work in IE7 browser. Here are examples:

http://jsfiddle.net/d_paul/3q96hw0b/3/ - with original code of _defaultGetter() method
http://jsfiddle.net/d_paul/xf4u8uhe/4/ - with modyfied code of _defaultGetter() method

Thank you for your help!
